discovery_engine_core/providers/src/expression.rs

```diff
@@ -94,8 +96,8 @@ impl Expr {
 
     pub(crate) fn build(self) -> String {
         match self {
-            Expr::And(exprs) => intersperse_exprs(exprs.into_iter(), " AND "),
-            Expr::Or(exprs) => intersperse_exprs(exprs.into_iter(), " OR "),
+            Expr::And(exprs) => intersperse_exprs(exprs.into_iter(), " && "),
+            Expr::Or(exprs) => intersperse_exprs(exprs.into_iter(), " || "),
```

Comment:
Is there any reason? They seems to support also the string variant

Reply:
You're right, from some of their examples it seemed like they only support the one and I didn't investigate further. Will revert that part.